.github/workflows/automate-review-labels.yml

```diff
@@ -48,12 +48,20 @@
                           repo: context.repo.repo,
                           labels: [label_no_staged, 'review-requested']
                       })
-                      github.rest.issues.removeLabel({
-                          issue_number: context.issue.number,
-                          owner: context.repo.owner,
-                          repo: context.repo.repo,
-                          name: ['Awaiting author contribution']
-                      });
+                      try {
+                        github.rest.issues.removeLabel({
+                            issue_number: context.issue.number,
+                            owner: context.repo.owner,
+                            repo: context.repo.repo,
+                            name: ['Awaiting author contribution']
+                        });
+                      } catch (e) {
+                        if (e.name == 'HttpError') {
+                         // Label doesn't exist.
+                        } else {
+                          throw e; // let other errors bubble up
+                        }
+                      };
```

Comment:
I'm not sure what `github.rest.issues.removeLabel` returns. We need to figure that out. If it's an asynchronous result like below, then we should use `.then()` and figure out what STATUS the server responds with instead of catching all `HttpError`.

Reply:
It looks like we want to ignore `status == 404` with message "label not found", but the server could also respond with `status = 410` which we we would not want to ignore, but it might also be raised as an `HttpError` in this API.

Reply:
- [ ] Also apply this patch to the other removeLabel call at the end of this file.

Reply:
Makes sense. One other approach could also be to split this up into two steps.

The first step adds the label and if the label can be added, sets a variable to true. That variable is then returned (Like above) and can be used for an `if` for the second step. The `if` will then also check `contains(github.event.issue.labels.*.name, 'Awaiting author contribution')`

Reply:
Yes. I'm a fan of the second step with conditional execution. Seems easier to read than error handling, but the conditional should have two conditions:
```yaml
if: >
  previous_step_found_team_mention
  and contains(github.event.issue.labels.*.name, 'Awaiting author contribution')
script: |
   // remove label
```

Reply:
Yes, that sounds good. We might need the `||` again for handling issue comments vs PRs but let me check if that's really the case.

Reply:
Have added that in https://github.com/conda-forge/staged-recipes/pull/18334/commits/937032331378243f7ce9631cdfdbd6b3a89a7878

Can you have a quick look if this makes sense and then I will copy it to the repo to test it.